### PR TITLE
Add CatBoosting regressor

### PR DESCRIPTION
**Pull request by underscrew**
### Status
**READY**

### Todo list
- [x] Documentation ??
- [x] Tests ??

### Description of the changes proposed in the pull request
Add [CatBoost Regressor](https://github.com/catboost/catboost)

### Where should the reviewer start?
Maybe testing the new regressor. Check for typos.

### CODE REVIEW

**Comment by underscrew** [comment]
I think all the linting is fixed now.

**Comment by underscrew** [comment]
@matheusfacure Done.

File: src/fklearn/training/regression.py [hunk 1]
```diff
@@ -186,6 +186,111 @@ def p(new_df: pd.DataFrame, apply_shap: bool = False) -> pd.DataFrame:
 xgb_regression_learner.__doc__ += learner_return_docstring("XGboost Regressor")
 
 
+@curry
+@log_learner_time(learner_name='cat_boost_regressor_learner')
+def CatBoostRegressor_learner(df: pd.DataFrame,
```
Comment (prcastro) [inline src/fklearn/training/regression.py]: Functions names in python are usually lowercase. Can you please change this to `catboost_regressor_learner` before we merge? Sorry about the delay on detecting this, should have pointed that on my first review.

Comment (underscrew) [reply src/fklearn/training/regression.py]: Done.

**Pull request merged**